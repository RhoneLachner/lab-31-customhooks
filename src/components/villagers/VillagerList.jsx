import React from 'react';
import PropTypes from 'prop-types';
import Villager from '../villagers/Villager';
import { Link } from 'react-router-dom';
import styles from '../containers/containers.css';

const VillagerList = ({ villagers }) => {
  const villagerElements = villagers.map(villager => (
    <div key={villager._id}>
      <Link to={`/details/${villager._id}`}>
        <Villager {...villager} />
      </Link>
    </div>
  ));

  return (
    <div data-testid="villagers" className={styles.villagerList}>
      {villagerElements}
    </div>
  );
};

VillagerList.propTypes = {
  villagers: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
    skill: PropTypes.string.isRequired,
    birthday: PropTypes.string.isRequired,
  })).isRequired
};

export default VillagerList;

