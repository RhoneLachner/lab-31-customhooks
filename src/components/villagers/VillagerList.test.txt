import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VillagerList from './VillagerList';

describe('VillagerList component', () => {
  afterEach(() => cleanup());
  it('renders VillagerList', () => {
    const { asFragment } = render(<VillagerList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
