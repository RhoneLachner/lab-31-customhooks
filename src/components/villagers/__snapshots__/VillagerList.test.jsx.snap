// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerList component renders VillagerList 1`] = `
<DocumentFragment>
  <div
    class="villagerList"
    data-testid="villagers"
  />
</DocumentFragment>
`;
